pkg/synchromanager/clustersynchro_manager.go

```diff
@@ -286,10 +297,23 @@ func (manager *Manager) reconcileCluster(cluster *clusterv1alpha2.PediaCluster)
 		}
 	}
 
+	if cluster.Spec.ShardingName == manager.shardingName {
+		if cluster.Status.ShardingName != nil && *cluster.Status.ShardingName != manager.shardingName {
```

Comment:
Judging this case here would result in an error deleting the cluster for which `status.sharding ! = name`, as there may be concurrent processing of the same cluster with other synchronization managers.

If a cluster is deleted by another cluster before the manager corresponding to status.sharding has started processing the cluster to be deleted, the cluster synchronizer may not be able to shut down, and dirty data may occur.